substrate/utils/build-script-utils/src/version.rs

```diff
@@ -59,3 +59,34 @@ fn get_version(impl_commit: &str) -> String {
 		impl_commit
 	)
 }
+
+/// Generate `SUBSTRATE_WASMTIME_VERSION`
+pub fn generate_wasmtime_version() {
+	generate_dependency_version("wasmtime", "SUBSTRATE_WASMTIME_VERSION");
+}
+
+fn generate_dependency_version(dep: &str, env_var: &str) {
+	// we only care about the root
+	match std::process::Command::new("cargo")
+		.args(["tree", "--depth=0", "--package", dep])
```

Comment:
```suggestion
		.args(["tree", "--depth=0", "--locked", "--offline", "--package", dep])
```

Reply:
Turns out that --locked alone would achieve both.